src/py/flwr/client/app.py

```diff
@@ -34,6 +34,8 @@ def start_client(
     client: Client,
     grpc_max_message_length: int = GRPC_MAX_MESSAGE_LENGTH,
     root_certificates: Optional[bytes] = None,
+    grpc_args: Optional[Iterable[Any]] = None,
```

Comment:
I'd only add specific arguments here instead of transparently passing all gRPC arguments. The client app should abstract connection handling for the user. For the specific use case you're addressing I'd only add an argument to toggle gRPC `wait_for_reay` defaulting to `True`.

Reply:
Great point. Adding only `wait_for_ready` would do just fine. IMO the parameter should default to `False` to preserve established behavior. I believe that by default, users expect the client to fail if a connection to the server cannot be established.

Reply:
Fair point! I was mostly considering the benefits and would have argued that these outdo the change in behavior but thinking about it again, I agree with you and it should default to `False`.